src/streaming/controllers/PlaybackController.js

```diff
@@ -106,13 +106,36 @@ function PlaybackController() {
         if (streamSwitch && isNaN(streamSeekTime)) return;
 
         // Seek new stream in priority order:
-        // - at seek time (streamSeekTime) that required period switch
+        // - at seek time (streamSeekTime) when switching period
         // - at start time provided in URI parameters
         // - at stream/period start time (for static streams) or live start time (for dynamic streams)
         let startTime = streamSeekTime;
         if (isNaN(startTime)) {
-            startTime = isDynamic ? e.liveStartTime : e.streamInfo.start;
-            startTime = Math.max(startTime, getStartTimeFromUri());
+            if (isDynamic) {
+                // For dynamic stream, start by default at (live edge - live delay)
+                startTime = e.liveStartTime;
+                // If start time in URI, take min value between live edge time and time from URI (if in DVR window range)
+                const dvrInfo = dashMetrics.getCurrentDVRInfo();
+                const dvrWindow = dvrInfo ? dvrInfo.range : null;
+                if (dvrWindow) {
+                    const startTimeFromUri = getStartTimeFromUriParameters(dvrWindow.start);
+                    if (!isNaN(startTimeFromUri) && startTimeFromUri >= dvrWindow.start) {
+                        logger.info('Start time from URI parameters: ' + startTimeFromUri);
```

Comment:
We might consider using dvrWindow.start if startTimeFromUri is < dvrWindow.start. I leave this open for discussion and will check if something about this is in the IOP

Reply:
Agree, this is an open question. There is nothing about that in DASH spec neither in IOP

Reply:
I did modify so that we use dvrWindow.start if startTimeFromUri is < dvrWindow.start